File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int> map;
        for(int i = 0;i<nums.size();i++){
            if(map.count(nums[i])){
                return true;
               
            }
            map[nums[i]] = 1;
        }
        return false;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        numsnew = sorted(nums)
        seen = set()
        result = []
        left = 1
        right = len(numsnew) - 1
        for i in range(len(numsnew)):
            if i > 0 and numsnew[i] == numsnew[i-1]: 
                continue
            left = i+1
            right = len(numsnew) - 1
            while left < right: 
                value = numsnew[i] + numsnew[left] + numsnew[right]
                if value > 0: 
                    right -= 1
                if value < 0:
                    left += 1
                if value == 0:

                    result.append([numsnew[i],numsnew[left],numsnew[right]])
                    while left < right and numsnew[left] == numsnew[left+1]: 
                        left += 1
                    while left < right and numsnew[right] == numsnew[right-1]: 
                        right -= 1
                    left += 1
                    right -= 1
        return result